:memo: Change hooks docs into a tutorial (#2094)
platformsh/platformsh-docs: docs/src/configuration/app/hooks/_index.md

   ```yaml {location="drupal/.platform.app.yaml"}
   hooks:
       build: |
           set -e
   ```

   There are two parts to the hook so far:

     * The `|` means the lines that follow can contain a series of commands.
       They aren't interpreted as new YAML properties.
     * Adding `set -e` means that the hook fails if _any_ of the commands in it fails.
       Without this setting, the hook fails only if its _final_ command fails.

       If a `build` hook fails for any reason, the build is aborted and the deploy doesn't happen.
       Note that this only works for `build` hooks.
       If other hooks fail, the deploy still happens.
1. Call your script from the `build` hook:

   ```yaml {location="drupal/.platform.app.yaml"}
   hooks:
       build: |
           set -e
           bash install-redis.sh 4.3.0
   ```

   This call includes the version of the Redis extension to compile.

Now the extension is compiled every time the Drupal app is built.

## Configure Drush

The template uses [Drush](https://www.drush.org/latest/) to handle routine tasks.
For its configuration, Drush needs the URL of the site.

> That means the configuration can't be done in the `build` hook, when the site isn't yet deployed and so the URL is unknown.

Reviewer: ```suggestion
That means the configuration can't be done in the `build` hook, when the site isn't yet deployed, because the URL is unknown at this stage.
```

Author: This suggestion shows that the original sentence is unclear. I wouldn't take the suggestion because the cause and effect are supposed to be the other way around (no deploy → no URL, not no URL → no deploy), but I will definitely try to clarify it.
